code/IntercomFormExtension.php

```diff
@@ -1,226 +1,235 @@
+     * @param  array $data           The array of mappings to update
+     * @return array
+     */
+    protected function addMappings($formField, $intercomField, $data)
+    {
+        $val = $this->owner->Fields()->dataFieldByName($formField)->dataValue();
+        if ($intercomField[0] === '$') {
+            if (!isset($data['custom_attributes'])) {
+                $data['custom_attributes'] = [];
+            }
+            $data['custom_attributes'][substr($intercomField, 1)] = $val;
+        } else {
+            $data[$intercomField] = $val;
+        }
+
+        return $data;
+    }
+
+    /**
+     * Sends the form data to Intercom, using the defined mappings
+     * @throws LogicException if some of the required configuration is missing
+     */
+    public function sendToIntercom()
+    {
+        if (empty($this->intercomFieldMapping)
+            && empty($this->intercomNoteMapping)
+            && empty($this->intercomCompanyFieldMapping)
+        ) {
+            throw new LogicException('You must define mapped fields to send a form submission to intercom, using Form::setIntercomFieldMapping() or Form::setIntercomNoteMapping()');
+        }
+
+        $intercom = Injector::inst()->get(Intercom::class);
+        $leadData = [];
+
+        foreach ($this->intercomUserFieldMapping as $formField => $intercomField) {
+            $leadData = $this->addMappings($formField, $intercomField, $leadData);
+        }
+
+        if (!empty($this->intercomCompanyFieldMapping)) {
+            $companyData = [];
+            foreach ($this->intercomCompanyFieldMapping as $formField => $intercomField) {
+                $companyData = $this->addMappings($formField, $intercomField, $companyData);
+            }
+            if (!isset($companyData['company_id'])) {
+                $companyData['company_id'] = time();
+            }
+
+            $leadData['companies'] = [$companyData];
+        }
+
+        try {
+            $this->owner->invokeWithExtensions('beforeSendToIntercom', $leadData);
+            $lead = $intercom->getClient()->createContact($leadData);
+
+            if (!empty($this->intercomNoteMapping)) {
+                $noteData = $this->intercomNoteHeader;
+                $noteData .= '<ul>';
+                foreach ($this->intercomNoteMapping as $fieldName => $label) {
+                    $noteData .= sprintf(
+                        '<li>%s: %s</li>',
+                        $label,
+                        $this->owner->Fields()->dataFieldByName($fieldName)->dataValue()
+                    );
+                }
+                $noteData .= '</ul>';
+
+                try {
+                    $intercom->getClient()->notes->create([
+                        'body' => $noteData,
+                        'user' => ['id' => $lead['id']]
+                    ]);
+                } catch (Exception $e) {
+                    Injector::inst()->get('Logger')->addWarning("Could not create note: {$e->getMessage()}");
```

Comment:
Needs to be updated to use `\Psr\Log\LoggerInterface::class` instead of `Logger` now